tests/test_modeling_common.py

```diff
@@ -1215,15 +1215,39 @@ def _create_and_check_torch_fx_tracing(self, config, inputs_dict, output_loss=Fa
                             (past_mask, inputs_to_test[1]["attention_mask"]), dim=1
                         )
 
-            if "inputs_embeds" in inspect.signature(model.forward).parameters and not model.config.is_encoder_decoder:
-                inputs_to_test.append(
-                    {
-                        "inputs_embeds": torch.rand(
-                            2, 2, model.config.hidden_size, dtype=torch.float, device=torch_device
-                        )
-                    }
+            if (
+                "input_ids" in inspect.signature(model.forward).parameters
+                and "inputs_embeds" in inspect.signature(model.forward).parameters
+                and not model.config.is_encoder_decoder
+            ):
```

Comment:
`is_encoder_decoder` is a cheaper check so should be at the start of the if logic statements 

```suggestion
            if (
                not model.config.is_encoder_decoder
                and "input_ids" in inspect.signature(model.forward).parameters
                and "inputs_embeds" in inspect.signature(model.forward).parameters
            ):
```